asv_bench/benchmarks/frame_methods.py

```diff
@@ -217,458 +215,325 @@ def time_isnull_obj(self):
-        self.df = DataFrame({'a': np.random.randint(((-1) << 8), (1 << 8), self.n), 'b': np.random.choice(self.t, self.n), 'c': np.random.choice(self.xs, self.n), })
-
-        self.df2 = DataFrame(np.random.randn(1000, 100).astype(str))
+        np.random.seed(1234)
+        n = (1 << 20)
+        t = date_range('2015-01-01', freq='S', periods=(n // 64))
+        xs = np.random.randn(n // 64).round(2)
+        self.df = DataFrame({'a': np.random.randint(-1 << 8, 1 << 8, n),
+                             'b': np.random.choice(t, n),
+                             'c': np.random.choice(xs, n)})
+        self.df2 = DataFrame(np.random.randn(1000, 100).astype(str)).T
 
     def time_frame_duplicated(self):
         self.df.duplicated()
 
     def time_frame_duplicated_wide(self):
-        self.df2.T.duplicated()
-
-
-
-
-
-
-
-
-
-
-
-
-
+        self.df2.duplicated()
 
 
+class XS(object):
 
-
-class frame_xs_col(object):
     goal_time = 0.2
+    params = [0, 1]
+    param_names = ['axis']
 
-    def setup(self):
-        self.df = DataFrame(randn(1, 100000))
-
-    def time_frame_xs_col(self):
-        self.df.xs(50000, axis=1)
-
-
-class frame_xs_row(object):
-    goal_time = 0.2
+    def setup(self, axis):
+        np.random.seed(1234)
+        self.N = 10**4
+        self.df = DataFrame(np.random.randn(self.N, self.N))
 
-    def setup(self):
-        self.df = DataFrame(randn(100000, 1))
+    def time_frame_xs(self, axis):
+        self.df.xs(self.N / 2, axis=axis)
 
-    def time_frame_xs_row(self):
-        self.df.xs(50000)
 
+class SortIndex(object):
 
-class frame_sort_index(object):
     goal_time = 0.2
+    params = [True, False]
+    param_names = ['ascending']
 
-    def setup(self):
-        self.df = DataFrame(randn(1000000, 2), columns=list('AB'))
+    def setup(self, ascending):
+        np.random.seed(1234)
+        self.df = DataFrame(np.random.randn(1000000, 2), columns=list('AB'))
 
-    def time_frame_sort_index(self):
-        self.df.sort_index()
+    def time_frame_sort_index(self, ascending):
+        self.df.sort_index(ascending=ascending)
 
 
-class frame_sort_index_by_columns(object):
+class SortIndexByColumns(object):
+
     goal_time = 0.2
 
     def setup(self):
-        self.N = 10000
-        self.K = 10
-        self.key1 = tm.makeStringIndex(self.N).values.repeat(self.K)
-        self.key2 = tm.makeStringIndex(self.N).values.repeat(self.K)
-        self.df = DataFrame({'key1': self.key1, 'key2': self.key2, 'value': np.random.randn((self.N * self.K)), })
-        self.col_array_list = list(self.df.values.T)
+        np.random.seed(1234)
+        N = 10000
+        K = 10
+        self.df = DataFrame({'key1': tm.makeStringIndex(N).values.repeat(K),
+                             'key2': tm.makeStringIndex(N).values.repeat(K),
+                             'value': np.random.randn(N * K)})
 
     def time_frame_sort_index_by_columns(self):
         self.df.sort_index(by=['key1', 'key2'])
```

Comment:
This is actually deprecated, I think we should do `sort_values` if that exists, otherwise, sort_index